src/tools/wasm-link.cc

```diff
@@ -625,6 +689,8 @@ static bool write_combined_section(Context* ctx,
   ctx->stream.WriteU8Enum(section_code, "section code");
   ctx->current_section_payload_offset = -1;
 
+  resolve_any_memory_import(ctx, sections);
```

Comment:
This needs to be moved somewhere where it only gets run once per output binary. It needs as input the memory sections and context, which was why I placed it here, but seems to be called far too many times.